lib/galaxy/datatypes/binary.py

```diff
@@ -1002,45 +1002,38 @@ def sniff(self, filename):
         False
         """
         if super(Biom2, self).sniff(filename):
-            try:
-                f = h5py.File(filename)
+            with h5py.File(filename, 'r') as f:
                 attributes = list(dict(f.attrs.items()))
                 required_fields = ['id', 'format-url', 'type', 'generated-by', 'creation-date', 'nnz', 'shape']
                 return set(required_fields).issubset(attributes)
-            except Exception:
-                return False
         return False
 
     def set_meta(self, dataset, overwrite=True, **kwd):
         super(Biom2, self).set_meta(dataset, overwrite=overwrite, **kwd)
-        try:
-            f = h5py.File(dataset.file_name)
+        with h5py.File(dataset.file_name, 'r') as f:
             attributes = dict(f.attrs.items())
 
-            dataset.metadata.id = attributes['id']
-            dataset.metadata.format_url = attributes['format-url']
+            dataset.metadata.id = util.unicodify(attributes['id'])
+            dataset.metadata.format_url = util.unicodify(attributes['format-url'])
             if 'format-version' in attributes:  # biom 2.1
-                dataset.metadata.format_version = '.'.join(map(str, list(attributes['format-version'])))
+                dataset.metadata.format_version = '.'.join(str(_) for _ in attributes['format-version'])
             elif 'format' in attributes:  # biom 2.0
-                dataset.metadata.format = attributes['format']
-            dataset.metadata.type = attributes['type']
-            dataset.metadata.shape = tuple(attributes['shape'])
-            dataset.metadata.generated_by = attributes['generated-by']
-            dataset.metadata.creation_date = attributes['creation-date']
+                dataset.metadata.format = util.unicodify(attributes['format'])
+            dataset.metadata.type = util.unicodify(attributes['type'])
+            dataset.metadata.shape = tuple((int(_) for _ in attributes['shape']))
+            dataset.metadata.generated_by = util.unicodify(attributes['generated-by'])
+            dataset.metadata.creation_date = util.unicodify(attributes['creation-date'])
             dataset.metadata.nnz = int(attributes['nnz'])
 
-        except Exception as e:
-            log.warning('%s, set_meta Exception: %s', self, e)
```

Comment:
We are a bit inconsistent in how we deal with exceptions in `set_meta` methods, should we just log them or let them bubble up?
See e.g. https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/datatypes/binary.py#L308

Reply:
I guess that depends case by case whether or not the dataset works correctly. I think I didn't mean to remove this one here though.

Reply:
OK, so if you restore it I think it's ready to merge.